roles/ceph-defaults/tasks/checks.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: test if devices are devices
+  parted:
+    device: "{{ item }}"
+    unit: MiB
+  with_items:
+    - "{{ devices }}"
+  register: device_exists
+  when:
+    - inventory_hostname in groups.get(osd_group_name, [])
+    - not osd_auto_discovery|default(False)
+    - osd_scenario|default('dummy') != 'lvm'
+
+- name: fail if one of the devices does not exist
+  fail:
+    msg: "{{ item.err }}"
+  with_items:
+    - "{{ device_exists.results }}"
+  when:
+    - inventory_hostname in groups.get(osd_group_name, [])
+    - not osd_auto_discovery|default(False)
+    - osd_scenario|default('dummy') != 'lvm'
+    - item.failed
+
+- name: test if dedicated_devices are devices
+  parted:
+    device: "{{ item }}"
+    unit: MiB
+  with_items:
+    - "{{ dedicated_devices }}"
```

Comment:
`dedicated_devices` is not always known when ceph-defaults runs, especially if running on collocated osd scenario.